File: patterns/embeddings/openai-like/javascript/basic-embeddings.ts
import 'dotenv/config';
import OpenAI from 'openai';

/**
 * Demonstrates generating text embeddings using the OpenAI API.
 */
async function main() {
    const apiKey = process.env.OPENAI_API_KEY;
    if (!apiKey) {
        console.error('Error: OPENAI_API_KEY not found in environment variables.');
        return;
    }

    const client = new OpenAI({
        apiKey: apiKey,
    });

    const textToEmbed = 'The quick brown fox jumps over the lazy dog.';
    console.log(`Generating embedding for: '${textToEmbed}'`);

    try {
        const response = await client.embeddings.create({
            input: textToEmbed,
            model: 'text-embedding-3-small',
        });

        const embedding = response.data[0].embedding;

        console.log(`\nEmbedding vector length: ${embedding.length}`);
        console.log(`First 5 dimensions: ${embedding.slice(0, 5)}`);
    } catch (error) {
        console.error('Error creating embedding:', error);
    }
}

main();

File: patterns/embeddings/openai-like/python/basic_embeddings.py
import os
from openai import OpenAI
from dotenv import load_dotenv

# Load environment variables
load_dotenv()

def main():
    """
    Demonstrates generating text embeddings using the OpenAI API.
    """
    api_key = os.environ.get("OPENAI_API_KEY")
    if not api_key:
        print("Error: OPENAI_API_KEY not found or empty.")
        return

    client = OpenAI(api_key=api_key)

    text_to_embed = "The quick brown fox jumps over the lazy dog."
    print(f"Generating embedding for: '{text_to_embed}'")

    response = client.embeddings.create(
        input=text_to_embed,
        model="text-embedding-3-small" # Efficient OpenAI embedding model
    )

    embedding = response.data[0].embedding
    
    print(f"\nEmbedding vector length: {len(embedding)}")
    print(f"First 5 dimensions: {embedding[:5]}")

if __name__ == "__main__":
    main()
